src/server/test.rs

```diff
@@ -31,13 +38,63 @@ fn test_endpoint(method: http::Method, path: &str, expected_body: &str) {
     assert_eq!(body, expected_body.as_bytes());
 }
 
+fn create_request(server: &TestServer, method: http::Method, path: &str) -> ClientRequest {
+    server
+        .client(method.clone(), path)
+        .set_header("Authorization", create_hawk_header(method.as_str(), path))
+        .finish()
+        .unwrap()
+}
+
+fn create_hawk_header(method: &str, path: &str) -> String {
+    let payload = HawkPayload {
+        expires: (Utc::now().timestamp() + 5) as f64,
+        node: "http://127.0.0.1:8000".to_string(),
+        salt: "wibble".to_string(),
+        uid: 42,
+    };
+    let payload = serde_json::to_string(&payload).unwrap();
+    let signing_secret = hkdf_expand_32(
+        b"services.mozilla.com/tokenlib/v1/signing",
+        None,
+        &[0u8; 32],
+    );
+    let mut signature: Hmac<Sha256> = Hmac::new_varkey(&signing_secret).unwrap();
+    signature.input(payload.as_bytes());
+    let signature = signature.result().code();
+    let mut id: Vec<u8> = vec![];
+    id.extend(payload.as_bytes());
+    id.extend_from_slice(&signature);
+    let id = base64::encode_config(&id, base64::URL_SAFE);
+    let token_secret = hkdf_expand_32(
+        format!("services.mozilla.com/tokenlib/v1/derive/{}", id).as_bytes(),
+        Some(b"wibble"),
+        &[0u8; 32],
+    );
+    let token_secret = base64::encode_config(&token_secret, base64::URL_SAFE);
+    let request = RequestBuilder::new(method, "127.0.0.1", 8000, path).request();
+    let credentials = Credentials {
+        id,
+        key: Key::new(token_secret.as_bytes(), &ring::digest::SHA256),
+    };
+    let header = request.make_header(&credentials).unwrap();
+    format!("Hawk {}", header)
```

Comment:
I opted for a manual implementation in the tests here, rather than leaning on code in `HawkPayload`, because I didn't want the possibility of a bug in production code leaking into the tests and preventing them from picking up on it.